### PR TITLE
Add trailing slash to job_history root.

### LINKED ISSUE #19721: Prow "Job History" Spammed with 0s pending runs
**What happened**:
Deck `Job History` is flooded with jobs that look like 
> <No Build ID>   0001-01-01 00:00:00 +0000 UTC     0s   <no result>

for [pull-kubernetes-e2e-gce](https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-e2e-gce)

image

**What you expected to happen**:
Only true runs show in job history.

**How to reproduce it (as minimally and precisely as possible)**:
View Job History link above.

**Please provide links to example occurrences, if any**:
https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-e2e-gce

**Anything else we need to know?**:

/cc @chaodaiG

### PR DESCRIPTION
**Pull request by vicsufer**
As noted in #19034 (comment) adding a trailing / to the job root fixes the issue.
After extracting the root field from the given URL, if it has no trailing `/` I add it.

fix #19721

/cc @MushuEE

### CODE REVIEW

File: prow/cmd/deck/job_history_test.go [hunk 1]
```diff
@@ -44,63 +44,63 @@ func TestJobHistURL(t *testing.T) {
...
-			root:            "logs/bar-e2e",
+			root:            "logs/bar-e2e/",
 			id:              emptyID,
 		},
 		{
 			address:         "http://www.example.com/job-history/s3/foo-bucket/logs/bar-e2e?buildId=123456789123456789",
 			bktName:         "foo-bucket",
 			storageProvider: providers.S3,
-			root:            "logs/bar-e2e",
+			root:            "logs/bar-e2e/",
```
Comment (MushuEE) [inline prow/cmd/deck/job_history_test.go:103]: Should we include a test where the root doesn't have `/` to cover your added `/` case?

Comment (vicsufer) [reply prow/cmd/deck/job_history_test.go:103]: Actually every test case checks that, the root is extracted from the `address` property, if it won't work as expected the result would be a root with no trailing `/`.

Comment (MushuEE) [reply prow/cmd/deck/job_history_test.go:103]: haha got it thanks

File: prow/cmd/deck/job_history.go [hunk 2]
```diff
@@ -307,6 +307,10 @@ func parseJobHistURL(url *url.URL) (storageProvider, bucketName, root string, bu
 		err = fmt.Errorf("invalid path for job: %v", url.Path)
 		return
 	}
+	// Append / to root if it does not end with it.
+	if !strings.HasSuffix(root, "/") {
```
Comment (alvaroaleman) [inline prow/cmd/deck/job_history.go:311]: @vicsufer please fill out the PR body and explain your fix. From what I see right now it has somewhere between no and a breaking effect:

* here its a no-op: https://github.com/kubernetes/test-infra/blob/7553d5a08c3361877bed3c3dfadb4bcf36740ac4/prow/cmd/deck/job_history.go#L427
* Here its again a no-op: https://github.com/kubernetes/test-infra/blob/7553d5a08c3361877bed3c3dfadb4bcf36740ac4/prow/cmd/deck/job_history.go#L237
* Here its breaking: https://github.com/kubernetes/test-infra/blob/7553d5a08c3361877bed3c3dfadb4bcf36740ac4/prow/cmd/deck/job_history.go#L251

/label tide/merge-method-squash

**Comment by alvaroaleman** [comment]
@vicsufer this change is way too broad. Please find out where exactly we are missing a trailing slash rather than this brute force method that may break other things along the way. In prow/cmd/deck/README.md is a description on how to run Deck locally.

**Comment by vicsufer** [comment]
@alvaroaleman you're correct, I'll try to find a workaround.

Should this PR be closed?

**Comment by alvaroaleman** [comment]
@vicsufer well whatever you prefer, updating this PR is also completely fine

**Pull request closed without merging**